### ISSUE TITLE
Column width incorrectly deduced on Windows (MSYS2)🐛 

### ISSUE THREAD

**Issue by legends2k** [dandavison/delta#1011]
Here's the raw output: `git --no-pager diff`:

```
   DEPENDS flatc
-  COMMENT "Generating header from schema IDL"
+  COMMENT "Generating savedata_generated.h from IDL savedata.fbs"
   VERBATIM)
```

Here's with delta as pager

```
│ 31 │  DEPENDS flatc                   │ 42 │  DEPENDS flatc
│ 32 │  COMMENT "Generating header fro↵ │    │
│    │m schema IDL"                     │    │
│    │                                  │ 43 │  COMMENT "Generating savedata_g↵
│    │                                  │    │enerated.h from IDL savedata.fbs"
│ 33 │  VERBATIM)                       │ 44 │  VERBATIM)
```

![image](https://user-images.githubusercontent.com/7668722/158411471-3d421eed-5e5f-4921-b98f-4c967a64ef47.png)

Here's the relevant snip from `.gitconfig`:

```
[core]
	pager = delta
[interactive]
	diffFilter = delta --color-only
[delta]
	features = side-by-side line-numbers decorations
	whitespace-error-style = 22 reverse
	syntax-theme = 1337
	navigate = true
[delta "decorations"]
	commit-decoration-style = brightgreen box ul
	file-style = bold yellow ul
	file-decoration-style = none
```

This is what I have in `.profile`

```
export LESS="$LESS --ignore-case --RAW-CONTROL-CHARS"
export DELTA_PAGER="$PAGER $LESS --quit-if-one-screen"
```

**Comment by th1000s** [comment]
This might be caused by [this terminal-size bug](https://github.com/eminence/terminal-size/issues/25) which in turn is used by `console`. The mentioned git bash shell is [MinGW-w64 / build under MSYS2](https://github.com/git-for-windows/MINGW-packages).

Given that `$COLUMNS` is correct and delta does not react to terminal resizing, a workaround should be possible. To that end, what is the output of `cargo run` inside an MSYS2 terminal after creating a project with `cargo init width_check` and setting the content of the files as follows?

<details><summary>Cargo.toml</summary>

```toml
[package]
name = "width_check"
version = "0.0.1"
edition = "2021"

[dependencies]
console = "0.15.0"
 ```
</details>

and 

<details><summary>src/main.rs</summary>

```rust
use console::Term;

fn main() {
    let term_stdout = Term::stdout();
    dbg!(term_stdout.is_term());
    dbg!(term_stdout.size());
}
 ```
</details>

**Comment by th1000s** [comment]
Yes, both `console` via `terminal-size` and `crossterm` seem to ultimately use `GetConsoleScreenBufferInfo()` on Windows which returns the fallback-value `(24, 79)` under MSYS2.

Can you try #1030 which calls `stty` (this seems to be available on MSYS2 and hopefully work better). The top commit contains a lot of debug output in case this does not work - but if it does you can revert it. Even with `--release` it might be a bit slower though since it will call the `stty` binary on startup.

**Comment by koprivajakub** [comment]
Is there any update on this, I can see @th1000s you mention the #1030 that should fix it but there is no release after this point. The tool is pretty much useless without the full width.

**Comment by Ceron257** [comment]
One thing I observed with git for windows: This issue is present when using git bash with mintty (the default). The `width_check` example always returns the fallback-value. 

When I use git bash with windows terminal the issue is **not** present👍🏼. And the `width_check` example also shows the correct values.

Hopefully this helps out anyone until we got a release with the fix.

**Comment by dandavison** [comment]
> The tool is pretty much useless without the full width.

@koprivajakub I'm sorry, I had absolutely no idea that delta was in an unusable state for any user! I had been thinking that width problems could always be overridden by command line options and/or env vars. In any case, I'll make a release today.
